### DISCUSSION TITLE
Is it possible to restrict, via roles, how many items a user can create

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#2976] [Archive]
Hi.

I want to allow users to create max X number of records for a specific collection, is that possible to configure?

**Comment by benhaynes** [comment]
Interesting — our permissions allow for this as of now, but it seems like a useful option to research. In the meantime, the way to accomplish this would be through an event hook ("create") with custom logic (if more than X items for owner, deny and throw exception).

**Reply by botzill** [reply]
Good point with the webhook. Will try that. Is the error from the webhook displayed on the page?

**Reply by benhaynes** [reply]
Yeah, you can throw an exception to do that:

https://github.com/directus/next/blob/main/docs/guides/extensions/api-hooks.md#before

**Reply by botzill** [reply]
OK, what should be the format of the error from the server webhook to display that on the page? Currently, we return 500 with some message  and on the page I see:

<img width="724" alt="Screen Shot 2020-11-12 at 16 57 29" src="https://user-images.githubusercontent.com/4686040/98955973-31c6a480-2508-11eb-9e8e-49c3ffb7ce45.png">

but no error message is showing on the page.

Please let me know the format of the error message or the code.

I do this like this now:

```
'items.create.before': async function (input, {event, accountability, collection, item, action, payload}) {
            try {
                await axios.post("${items_create_before}", {
                    event,accountability,collection,item,action,payload
                });
            } catch (error) {
                throw new InvalidPayloadException(error.message);
            }
            return input
        },
```
Thx.
